docs/articles/vector_search_based_recommender_system.md

```diff
@@ -0,0 +1,276 @@
+os.environ["ROCKSET_API_KEY"] = "XveaN8L9mUFgaOkffpv6tX6VSPHz####"
+```
+
+### Step 2: Create a new Collection and Upload Data
+
+For this tutorial, we'll be using Amazon product review [data](https://drive.google.com/file/d/1EEvCUqKIH6LuLLGRjo6ayCEXqT4Z-ZVE/view?usp=drive_link). Download this on your local machine so it can be uploaded to your collection.
+
+We'll be uploading the [sample_data.json](https://drive.google.com/file/d/1EEvCUqKIH6LuLLGRjo6ayCEXqT4Z-ZVE/view?usp=drive_link) file to Rockset.
+
+Note: In practice, the data is usually ingested from another streaming service but for keeping things simple and building a demo application we are using a sample from a public dataset.
+
+### Step 3: Create OpenAI API Key
+
+To convert data into [embeddings](https://platform.openai.com/docs/api-reference/embeddings/create), we'll use OpenAI's model.
+
+After signing up, go to [API Keys](https://platform.openai.com/api-keys) and create a secret key. Don't forget to copy and save your key that will look similar to "sk-***********************". Like Rockset's API key, save your OpenAI key in the environment so it can easily used throughout the code:
+
+```python
+import os
+os.environ["OPENAI_API_KEY"] = "sk-####"
+```
+### Step 4: Create a Query Lambda on Rockset
+
+[Query Lambdas](https://docs.rockset.com/documentation/docs/query-lambdas) are named, parameterized SQL queries stored in Rockset that can be executed from a dedicated REST endpoint. Using Query Lambdas, you can save your SQL queries as separate resources and manage them successfully through development and production.
+
+Let's create one for our tutorial. We'll be using the following Query Lambda with parameters: embedding, brand, min_price, max_price and limit.
+
+```sql
+SELECT
+  asin,
+  title,
+  brand,
+  description,
+  estimated_price,
+  brand_tokens,
+  image_ur1,
+  APPROX_DOT_PRODUCT(embedding, VECTOR_ENFORCE(:embedding, 1536, 'float')) as similarity
+FROM
+    commons.sample s
+WHERE estimated_price between :min_price AND :max_price
+AND ARRAY_CONTAINS(brand_tokens, LOWER(:brand))
+ORDER BY similarity DESC
+LIMIT :limit;
+```
+This parameterized query does the following:
+- It retrieves data from the "sample" table in the "commons" schema. And selects specific columns like ASIN, title, brand, description, estimated_price, brand_tokens, and image_ur1.
+- It also computes the similarity between the provided embedding and the embedding stored in the database using the **APPROX_DOT_PRODUCT** function. 
+- The query filters results based on the estimated_price falling within the provided range, the brand containing the specified value, and then sorts the results based on similarity in descending order.
+- Finally, it limits the number of returned rows based on the provided limit parameter.
+
+
+To build this Query Lambda, query the collection made in step 2 by clicking on **Query this collection** and pasting the parameterized query above into the Rockset query editor.
+
```

Comment:
Where do users find **Query this collection**?

Reply:
in the last image of step 2 as mentioned over here. It's a button visible in step 2's picture.

Reply:
@fatirmalik1 
Unless I'm missing something, "### Step 2: Create a new collection, and upload data" has no image.
You have three images linked in the article (image1.png, image2,png, image3.png) and none of these have a button called Query this collection. Can you point me more directly to what you're referring to?

Reply:
I don't see this in the doc I gave, @fatirmalik1  why do we have this in md file, the steps to create a collection were deleted because we didn't wanna keep low-level Rockset signup details. 

@robertdhayanturner  feel free to refer to the doc: https://docs.google.com/document/d/1Od4OWG0t6wjntFvkT_hT9Ap2ygyJi8xW99jrpQcCZV4/edit?usp=sharing and modify the md as you see fit.

Reply:
@ankit1khare the md has verbatim the same wording as the doc you linked.  There is no 'Query this collection' button. Still need clarification on this point.